_posts/2020-07-12-ray-tune.md

```diff
@@ -0,0 +1,457 @@
+> 1. **Consistency with Scikit-Learn API** ...
+> 2. **Modern hyperparameter tuning techniques** ...
+> 3. **Framework support** ...
+> 4. **Scale up** ...
+>
+> [Ray's] Tune-sklearn is also **fast**.
+
+Dask-ML's model selection module has every one of the features. Let's look at
+each feature and see what Dask-ML and Ray have to offer:
+
+### Consistency with the Scikit-learn API
+
+*Dask-ML is also consistent with the Scikit-learn API.*
+
+Here's how to use Scikit-learn's, Dask-ML's and Ray's tune-sklearn
+hyperparameter optimization:
+
+``` python
+from scipy.stats import loguniform
+from sklearn.datasets import make_classification
+from sklearn.linear_model import SGDClassifier
+from dask.distributed import Client
+client = Client()
+
+X, y = make_classification()
+model = SGDClassifier()
+params = {"loss": ["hinge", "log", "perceptron"],
+          "alpha": loguniform(1e-5, 1e-3)}
+
+# Scikit-learn hyperparameter optimization
+from sklearn.model_selection import RandomizedSearchCV
+search = RandomizedSearchCV(clf, params, ...)
+search.fit(X, y)
+
+# Dask hyperparameter optimization
+from dask_ml.model_selection import HyperbandSearchCV
+search = HyperbandSearchCV(model, params, ...)
+search.fit(X, y, classes=[0, 1])
+
+# Ray hyperparameter optimization
+from tune_sklearn import TuneSearchCV
+search = TuneSearchCV(clf, params, ...)
+search.fit(X, y, classes=[0, 1])
+```
+
+Clearly, both Dask-ML and Ray's tune-sklearn are Scikit-learn compatible. Now
+let's focus on how each search performs and how it's configured.
+
+[skl-eg]:https://scikit-learn.org/stable/auto_examples/model_selection/plot_randomized_search.html#sphx-glr-auto-examples-model-selection-plot-randomized-search-py
+[sgd-eg]:https://github.com/ray-project/tune-sklearn/blob/31f228e21ef632a89a74947252d8ad5323cbd043/examples/sgd.py
+
+### Modern hypereparameter tuning techniques
+
+Ray's tune-sklearn provides access to advanced hyperparameter optimization, and
+[the introduction][1] made this claim:
+
+> tune-sklearn is the only
+> Scikit-Learn interface that allows you to easily leverage Bayesian
+> Optimization, HyperBand and other optimization techniques by simply toggling a few parameters.
+
+*Dask-ML also offers state-of-the-art hyperparameter tuning techniques that
+are easily accessible.*
+
+Methods to avoid unnecessary work in hyperparameter optimization include the
+following include the following:
+
+1. Stopping poor scoring models before they complete training.
+2. Using prior model training to figure out the best set of hyperparameters.
+
+Dask-ML focuses on the first option because it's easier to parallelize.
+Dask-ML has an implementation of "Hyperband," an early stopping technique for
+Scikit-learn's `RandomizedSearchCV`. Hyperband and `RandomizedSearchCV` are the
+same, except Hyperband makes decisions on when to stop models.[^stopping]
+
+Hyperband is popular because it works.  After the introduction of Hyperband in
```

Comment:
Can this be omitted?

Our focus should be on correcting the inaccurate claims, not explaining details of algorithms.  This feels a bit extraneous.